File: webapp/src/app/devices/DevicesComponent.scala
package org.linnando.mixemulator.webapp

import angulate2.std._

import scala.scalajs.js

@Component(
  selector = "devices",
  templateUrl = "webapp/src/main/resources/devices.component.html",
  styleUrls = @@@("webapp/src/main/resources/devices.component.css")
)
class DevicesComponent(virtualMachineService: VirtualMachineService) extends OnInit {
  var deviceNumbers: js.Array[Int] = js.Array()
  var selectedDevice: Int = _

  override def ngOnInit(): Unit = {
    deviceNumbers = js.Array[Int]() ++ virtualMachineService.deviceNumbers
  }

  def deviceDescription(deviceNumber: Int): String = {
    val deviceName = virtualMachineService.deviceName(deviceNumber)
    s"$deviceNumber - $deviceName"
  }
}

File: webapp/src/app/devices/devices.component.ts
import { Component, OnInit } from '@angular/core';
import { VirtualMachineService } from '../virtual-machine.service';

@Component({
  selector: 'mix-devices',
  templateUrl: './devices.component.html',
  styleUrls: ['./devices.component.scss']
})
export class DevicesComponent implements OnInit {
  deviceNumbers: number[] = [];
  selectedDevice = 0;

  constructor(private virtualMachineService: VirtualMachineService) { }

  ngOnInit(): void {
    this.deviceNumbers = [...this.virtualMachineService.deviceNumbers];
  }

  deviceDescription(deviceNumber: number): string {
    const deviceName = this.virtualMachineService.deviceName(deviceNumber);
    return `${deviceNumber} - ${deviceName}`;
  };
}
